lib/openstack_api/openstack_project.py

```diff
@@ -18,20 +18,30 @@ def create_project(
     :param project_details: A datastructure containing all the project details
     :return: A clouds project, or None if it was unsuccessful
     """
-    # domain_id can be blank, we will create a project in the default domain
+
     if not project_details.name:
         raise MissingMandatoryParamError("The project name is missing")
     if not project_details.email:
         raise MissingMandatoryParamError("The project contact email is missing")
     email_regex = r"^[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$"
     if not re.match(email_regex, project_details.email):
         raise ValueError("The project contact email is invalid")
+    if not project_details.domain:
+        raise MissingMandatoryParamError("The domain for the project is missing")
 
     create_project_kwargs = {
         "name": project_details.name,
         "description": project_details.description,
         "is_enabled": project_details.is_enabled,
     }
+
+    # get domain id for the domain to use
+    domain_name = project_details.domain
+    domain_details = conn.identity.get_domain(domain_name)
+    domain_id = domain_details.id
+
+    create_project_kwargs["domain_id"] = domain_id
```

Comment:
I think this might be simpler as we are not reusing these variables elsewhere.
```suggestion
    domain_details = conn.identity.get_domain(project_details.domain)

    create_project_kwargs["domain_id"] = domain_details.id
```

Reply:
Good spot, I was meaning to simplify the line but forgot when writing the unit test for it 😅